### PR TITLE
[12.0][WIP][ADD] fieldservice_contract

### PR DESCRIPTION
**Pull request by hparfr**
The idea here is to manage the recurring orders from a contract.
Contracts in v12 are lighter and easier to use than agreements. 

The common use case will be the following:
A sale order creates a contract,
The contract generates field services recurring orders.
The contract will take care of re-conduction, and invoicing per period.

 Lead (crm) --> Sale Order --> Contract --> Recurring Field Service -> FSM Orders
                                                                   --> Invoices (where qty = FSM Orders within period)

Still work in progress.

Some parts are based on fieldservice_sale. And the both of them play nicely together.

### CODE REVIEW

File: fieldservice_contract/__manifest__.py
Comment (brian10048) [inline fieldservice_contract/__manifest__.py:14]:
```suggestion
        'contract',
        'product_contract',
```

Need to add this dependency for sale order to create the contract

File: fieldservice_contract/__manifest__.py
Comment (brian10048) [inline fieldservice_contract/__manifest__.py:16]:
```suggestion
        'fieldservice_sale',
        'fieldservice_sale_recurring',
```

Need to add this dependency since #371 split up fieldservice_sale

**Review by brian10048 — CHANGES_REQUESTED** [review]
Thank you @hparfr!  

Is this still a WIP?

I made a few suggestions in the code review based on what I found during testing. 

Also, can you add the readme directory with its files and include a way to install this from the base fieldservice module in res.config.settings?

After that this looks ready to merge unless you have further improvements to make.

**Comment by hparfr** [comment]
Thanks @brian10048 ,
Yes it's kinda still a wip. I need to check one or two functional things, the rest is improve code quality (add readme, comments, fix your remarks, etc.).

**Comment by hparfr** [comment]
I'm waiting, this PR to be accepted https://github.com/brian10048/field-service/pull/8/files
Then this one : https://github.com/OCA/field-service/pull/403
Then, I need to push a refactoring commit.
and it will be ready to be merged :)

**Comment by brian10048** [comment]
> 
> I'm waiting, this PR to be accepted https://github.com/brian10048/field-service/pull/8/files
> Then this one : #403
> Then, I need to push a refactoring commit.
> and it will be ready to be merged :)

@hparfr. These are now merged in case you had not noticed.  
Let me know when you'd like me to review this and I'll give it a test!  Looking forward to it

**Comment by hparfr** [comment]
Replaced by #809 (v14)

**Pull request closed without merging**
